### PR TITLE
Build: Add KUBE_GCS_RELEASE_BUCKET_MIRROR option to push-ci-build.sh

### PR DESCRIPTION
**Pull request by zmerlynn**
And start pushing to new CI bucket as well.

(This will need a cherrypick back to `release-1.3` before I can do any bucket surgery, FWIW.)

[Analytics]()

### CODE REVIEW

**Comment by zmerlynn** [comment]
See https://groups.google.com/d/msg/kubernetes-dev/2_71FD_gScM/bSzMZfWVCAAJ for context (and an internal thread).

cc @kubernetes/test-infra-maintainers

**Comment by ixdy** [comment]
cc @kubernetes/release-maintainers

**Comment by roberthbailey** [comment]
lgtm

File: build/common.sh [hunk 1]
```diff
@@ -1229,6 +1229,16 @@ function kube::release::gcs::copy_release_artifacts() {
   fi
 
   gsutil ls -lhr "${gcs_destination}" || return 1
+
+  if [[ -n "${KUBE_GCS_RELEASE_BUCKET_MIRROR:-}" ]]; then
+    local -r gcs_mirror="gs://${KUBE_GCS_RELEASE_BUCKET_MIRROR}/${KUBE_GCS_RELEASE_PREFIX}"
+    kube::log::status "Mirroring build to ${gcs_mirror}"
+    gsutil -q -m "${gcs_options[@]+${gcs_options[@]}}" rsync -d -r "${gcs_destination}" "${gcs_mirror}"
```
Comment (ixdy) [inline build/common.sh]: maybe add `|| return 1` ?

File: build/common.sh [hunk 2]
```diff
@@ -1493,7 +1503,19 @@ function kube::release::gcs::verify_ci_ge() {
 #   If new version is greater than the GCS version
 function kube::release::gcs::publish() {
   local -r publish_file="${1-}"
-  local -r publish_file_dst="gs://${KUBE_GCS_RELEASE_BUCKET}/${publish_file}"
+
+  kube::release::gcs::publish_to_bucket "${KUBE_GCS_RELEASE_BUCKET}" "${publish_file}"
```
Comment (ixdy) [inline build/common.sh]: should this return on failure to match previous behavior?

File: build/common.sh (hunk 1 above)
Comment (zmerlynn) [inline build/common.sh]: Yup. :)

File: build/common.sh (hunk 2 above)
Comment (zmerlynn) [inline build/common.sh]: Fixed.

**Comment by zmerlynn** [comment]
PTAL

**Comment by ixdy** [comment]
LGTM

**Comment by zmerlynn** [comment]
@k8s-oncall: Requesting manual merge so I can babysit `kubernetes-build` job during business hours.

**Pull request merged**